### ISSUE TITLE
PropertySet.detach is broken

### ISSUE THREAD

**Issue by rymsha** [enonic/xp#10571]
This method is deprecated and does not have tests

Simple test fails
```
    @Test
    void detach()
    {
        PropertySet set = new PropertyTree().newSet();
        set.setString( "myProp", "myValue" );

        PropertySet detached = set.detach(); //<- Exception here
    }
```
